SPECIFICATION.md

```diff
@@ -184,60 +185,60 @@ JSON example:
-A **DOSE** is an array containing a dose ID, a vaccine producer designation, and a lot number. Example: `[1, "PFIZER", "13a056"]` indicates "Dose 1, from Pfizer, lot number 13a056." In the event of the Vaccine Producer Designation exceeding the storage capacity of SHORTSTRING, only the first eight (8) bytes of the Vaccine Producer Designation should be used.
-Fields:
-1. Dose ID: **SHORTNUMERIC**.
-1. Vaccine Producer Designation: **SHORTSTRING**.
-1. Lot number: **SHORTSTRING**.
-
-#### **DOSE** Serialization
-1. For URI formats, the **DOSEINFO** structure must be serialized to a string.
-1. This serialization should be accomplished by joining each serialized **DOSE**
-structure inside the **DOSEINFO** structure.
-1. The elements MUST be concatenated in order with the ctrl-] (character code 29, hex 1D, GS, or Group Separator) delimiter.
-Pseudo-code:
-```
-serialize(DOSE) ::= join("\x1D", [1, "PFIZER", "13a056"])
--> "1\x1DPFIZER\x1D13a056"
-```
+1. *manuf*: **SHORTSTRING**. The name of the manufacturer of the vaccine
+1. *product*: **SHORTSTRING**. The name of the product of the vaccine. 
+1. *lot*: **SHORTSTRING**. The lot number of bottle of the vaccine. 
+1. *route* (optional): **SHORTSTRING**. The route of application. Options are: 
+   1. C38238	INTRADERMAL
+   1. C28161	INTRAMUSCULAR
+   1. C38276	INTRAVENOUS
+   1. C38284	NASAL
+   1. C38288	ORAL
+   1. C38676	PERCUTANEOUS
+   1. C38299	SUBCUTANEOUS
+   1. C38305	TRANSDERMAL
+1. *site* (optional): **SHORTSTRING**. The site of the application. Options are: 
+   1. LA:	Left Arm
+   1. LD:	Left Deltoid
+   1. LG:	Left Gluteus Medius
+   1. LLFA:	Left Lower Forearm
+   1. LT:	Left Thigh
+   1. LVL:	Left Vastus Lateralis
+   1. RA:	Right Arm
+   1. RD:	Right Deltoid
+   1. RG:	Right Gluteus Medius
+   1. RLFA:	Right Lower Forearm
+   1. RT:	Right Thigh
+   1. RVL:	Right Vastus Lateralis
+1. *dose* (optional): **NUMERIC**. A dose size in ml. 
+1. *boosts*: An array of **SHORTNUMERIC** representing the distance in days from the first dose. Ex, for Moderna's (two doses): [28], for JnJ (just one dose): []
+1. *passkey*: **STRING**. The cryptographic hash of the data in the Passkey, as defined in the Passkey specification.
 
 ### Badge Serialization Order:
-In situations requiring payload serialization, the fields in the Badge payload MUST be serialized in the following order:
-1. coupon
-2. doseinfo
-3. passkey
+In situations requiring data serialization, the fields in the Coupon payload MUST be serialized in the following order:
+1. Manuf
+1. Product
+1. Lot
+1. Route
+1. Site
+1. dose
+1. boosts
+1. passkey
 
 JSON example:
 ```json
 {
   "type": "badge",
   "version": 1,
   "data": {
-    "coupon": "5a688b8230705d88b9f3bef1f23e099f8ee140e04c05a8575531808810019487",
-    "doseInfo": [[1, "PFIZER", "13a056"], [2, "PFIZER", "29a063"]],
+    "manuf" : "Moderna",
+    "product" : "Covid19",
+    "lot": ":23092",
+    "route": "RA",
+    "site": "C28161",
+    "dose": 0.5,
```

Comment:
NUMERIC does not support floats (no decimal). Can we specify the dosage in a different unit (microliters perhaps)?